importer/src/module_loader.rs

```diff
@@ -25,6 +65,47 @@ type Error = String;
 impl<T: FieldElement> Folder<T> for Loader {
     type Error = Error;
 
+    fn fold_module_value(&mut self, module: ASMModule<T>) -> Result<ASMModule<T>, Self::Error> {
+        // Fold module statements. This code is identical to the default implementation of Folder::fold_module_value
+        // (but there is no way of accessing it here...)
+        let mut statements = module
+            .statements
+            .into_iter()
+            .map(|s| match s {
+                ModuleStatement::SymbolDefinition(d) => match d.value {
+                    SymbolValue::Machine(machine) => self.fold_machine(machine).map(From::from),
+                    SymbolValue::Import(import) => {
+                        <Loader as Folder<T>>::fold_import(self, import).map(From::from)
+                    }
+                    SymbolValue::Module(module) => self.fold_module(module).map(From::from),
+                }
+                .map(|value| ModuleStatement::SymbolDefinition(SymbolDefinition { value, ..d })),
+            })
+            .collect::<Result<Vec<_>, _>>()?;
+
+        // The current module uses the standard library if any import starts with `std::`
+        let uses_std = statements.iter().any(|s| {
```

Comment:
Is this reliable? Shouldn't we just always include it? What if you do some weird things and import it via ``super::std`` or something? Is that possible?

Reply:
True, that would break (unless the parent module happens to use `std`), but maybe that's fine? I's say `std` should always be imported as an absolute import.

The trouble with always including it is that the same code is used to load the STD itself. Always including it would lead to an infinite loop. This could be resolved though by having a different loader for the STD (one that only has a custom `fold_module` and the default `fold_module_value`). Happy to do that if you think it's better.

Reply:
More a `mod std;` (with the right path as context), but yes.

I just refactored the code. Now the `std` module is added unconditionally to all modules in a separate pass. I agree it's much cleaner! 🎉